packages/memory-storage/src/resource-clients/request-queue.ts

```diff
@@ -163,6 +175,98 @@ export class RequestQueueClient extends BaseClient implements storage.RequestQue
         };
     }
 
+    async listAndLockHead(options: storage.ListAndLockOptions): Promise<storage.ListAndLockHeadResult> {
+        const { limit, lockSecs } = s.object({
+            limit: s.number.optional.default(100),
+            lockSecs: s.number,
+        }).parse(options);
+
+        const queue = await this.getQueue();
+
+        const start = Date.now();
+        const isLocked = (request: InternalRequest) => !request.orderNo || request.orderNo > start || request.orderNo < -start;
+
+        const items = [];
+        for (const storageEntry of queue.requests.values()) {
+            if (items.length === limit) {
+                break;
+            }
+
+            const request = await storageEntry.get();
+
+            if (isLocked(request)) {
+                continue;
+            }
+
+            request.orderNo = (start + lockSecs * 1000) * (request.orderNo! > 0 ? 1 : -1);
+            await storageEntry.update(request);
+
+            items.push(request);
+        }
+
+        return {
+            limit,
+            lockSecs,
+            hadMultipleClients: false,
+            queueModifiedAt: queue.modifiedAt,
+            items: items.map(({ json }) => this._jsonToRequest(json)!),
+        };
+    }
+
+    async prolongRequestLock(id: string, options: storage.ProlongRequestLockOptions) : Promise<storage.ProlongRequestLockResult> {
+        s.string.parse(id);
+        const { lockSecs, forefront } = s.object({
+            lockSecs: s.number,
+            forefront: s.boolean.optional.default(false),
+        }).parse(options);
+
+        const queue = await this.getQueue();
+        const request = await queue.requests.get(id)?.get();
```

Comment:
Maybe instead of calling get here, and lower on line 240, you can cache the entry in its own variable 👀

Reply:
nice catch, thanks!